ballerina/schema-client.bal

```diff
@@ -0,0 +1,319 @@
+    # + return - An `sql:Error` or `()`
+    public function init(string host, string user, string password, string database, int port, 
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {
+        self.database = database;
+        self.dbClient = check new (host, user, password, database, port, options, connectionPool);
+    }
+
+    # Retrieves all tables in the database.
+    #
+    # + return - A string array containing the names of the tables or an `sql:Error`
+    isolated remote function listTables() returns string[]|sql:Error {
+        string[] tables = [];
+        stream<record {}, sql:Error?> results = self.dbClient->query(
+            `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES 
+             WHERE TABLE_SCHEMA = ${self.database};`
+        );
+
+        do {
+            tables = check from record {} result in results
+                select <string>result["TABLE_NAME"];
+        } on fail error e {
+            return error sql:Error(string `Error while listing the tables in the ${self.database} database.`, cause = e);
+        }
+
+        check results.close();
+
+        return tables;
+    }
+
+    # Retrieves information relevant to the provided table in the database.
+    #
+    # + tableName - The name of the table
+    # + include - Options on whether column and constraint related information should be fetched.
+    #             If `NO_COLUMNS` is provided, then no information related to columns will be retrieved.
+    #             If `COLUMNS_ONLY` is provided, then columnar information will be retrieved, but not constraint
+    #             related information.
+    #             If `COLUMNS_WITH_CONSTRAINTS` is provided, then columar information along with constraint related
+    #             information will be retrieved
+    # + return - An 'TableDefinition' with the relevant table information or an `sql:Error`
+    isolated remote function getTableInfo(string tableName, sql:ColumnRetrievalOptions include = sql:COLUMNS_ONLY) returns TableDefinition|sql:Error {
+        record {}|sql:Error 'table = self.dbClient->queryRow(
+            `SELECT TABLE_TYPE FROM INFORMATION_SCHEMA.TABLES 
+             WHERE (TABLE_SCHEMA=${self.database} AND TABLE_NAME = ${tableName});`
+        );
+
+        if 'table is sql:NoRowsError {
+            return error sql:NoRowsError("The selected table does not exist or the user does not have the required privilege level to view the table.");
+        } else if 'table is sql:Error {
+            return 'table;
+        } else {
+            TableDefinition tableDef = {
+                name: tableName,
+                'type: <sql:TableType>'table["TABLE_TYPE"]
+            };
+
+            if !(include == sql:NO_COLUMNS) {
+                TableDefinition|sql:Error tableDefError = self.getColumns(tableName, tableDef);
```

Comment:
It would be better if we can change the `getColumns()` method to return `sql:ColumnDefinition[]`.

```suggestion
                sql:ColumnDefinition[] columnDefinitions = self.getColumns(tableName);
                tableDef.columns = columnDefinitions;
```